src/elm/Pages/Pipeline/View.elm

```diff
@@ -221,35 +236,58 @@ wrapPipelineConfigurationContent model cls content =
         body
 
 
-{-| viewTemplatesExpansion : takes model and renders the config header button for expanding pipeline templates.
+velaYmlFileName =
+    "vela.yml"
+
+
+{-| viewPipelineActions : takes model and renders the config header buttons for expanding pipeline templates and downloading yaml.
 -}
-viewTemplatesExpansion : PartialModel a -> Html Msg
-viewTemplatesExpansion model =
-    case model.templates of
-        ( Success templates, _ ) ->
-            if Dict.size templates > 0 then
-                div [ class "expand-templates", Util.testAttribute "pipeline-templates-expand" ]
-                    [ expandTemplatesToggleIcon model.pipeline
-                    , expandTemplatesToggleButton model.pipeline
-                    , expandTemplatesTip
-                    ]
+viewPipelineActions : PartialModel a -> Get msg -> Expand msg -> Download msg -> Html msg
+viewPipelineActions model get expand download =
+    let
+        t =
+            case model.templates.data of
+                Success templates ->
+                    if Dict.size templates > 0 then
+                        div [ class "action", class "expand-templates", Util.testAttribute "pipeline-templates-expand" ]
+                            [ expandTemplatesToggleButton model.pipeline get expand
+                            , expandTemplatesToggleIcon model.pipeline
+                            , expandTemplatesTip
+                            ]
 
-            else
-                text ""
+                    else
+                        text ""
 
-        ( Loading, _ ) ->
-            text ""
+                _ ->
+                    text ""
 
-        _ ->
-            text ""
+        d =
```

Comment:
adding a "download vela.yml" button

this could have easily been in a separate PR but it utilizes the newly shared `DownloadFile` msg so it was appropriate to sneak in this PR.